File: ui/app/lib/types/state.ts
export interface ContentItem {
  lc: number;
  type: string;
  id: string[];
  repr: string;
}

export interface AgentState {
  messages: { content: string }[];
  content_items: ContentItem[];
  generated_posts: string[];
  logs?: { message: string; done: boolean }[];
}

export interface ContentUpdateHandler {
  onContentUpdate: (index: number, newContent: string) => void;
} 

File: agent/backend/schema/state.py
from typing import Annotated, List, Optional

from pydantic import BaseModel, Field
from langchain_core.messages import (
    AnyMessage,
)
from backend.schema.schema import (
    ContentItem,
    Log,
)
from langgraph.graph.message import add_messages




class AgentState(BaseModel):
    messages: Annotated[List[AnyMessage], add_messages] = Field(default_factory=list)
    next_action: Annotated[Optional[str], Field(default=None)]
    content_items: Annotated[Optional[List[ContentItem]], Field(default=None)]
    generated_posts: Annotated[Optional[List[str]], Field(default=None)]
    writer_examples: Annotated[Optional[List[str]], Field(default=None)]
    logs: list[Log] = Field(default=[])
